pam/internal/gdm/conversation.go

```diff
@@ -26,3 +29,138 @@ func sendToGdm(pamMTx pam.ModuleTransaction, data []byte) ([]byte, error) {
 	defer binRes.Release()
 	return binRes.Decode(decodeJSONProtoMessage)
 }
+
+// sendData sends the data to the PAM Module, returning the JSON data.
+func sendData(pamMTx pam.ModuleTransaction, d *Data) ([]byte, error) {
+	bytes, err := d.JSON()
+	if err != nil {
+		return nil, err
+	}
+
+	return sendToGdm(pamMTx, bytes)
+}
+
+// SendData sends the data to the PAM Module and returns the parsed Data.
+func SendData(pamMTx pam.ModuleTransaction, d *Data) (*Data, error) {
+	bytes, err := d.JSON()
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf(context.TODO(), "Sending to GDM: %s", bytes)
+	jsonValue, err := sendToGdm(pamMTx, bytes)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
You should then call the private one inside the public one, and then we can cut the first lines of it and justify the existence of the private one.

Reply:
Oh, sure thing... That was supposed to be in this way, for some reason I didn't do it.... That's done now.